edgelet/kube-client/src/client.rs

```diff
@@ -809,27 +791,51 @@ where
                 Error::from(err.context(ErrorKind::UrlJoin(self.config.host().clone(), path)))
             })
             .and_then(|url| {
-                // set the full URL on the request including API path
-                *req.uri_mut() = url.as_str().parse::<Uri>().context(ErrorKind::Uri(url))?;
+                // req is an http 0.2 Request but hyper uses http 0.1, so destructure req and reassemble it.
```

Comment:
Question - Is there advantages to switching to hyper (which seems to use an older version of http - 0.1)?

Reply:
No, it's just to have a reference to http 0.1. The alternative would've been to add a dep on http:0.1, remove the `k8s_openapi::http` import, and change all uses of the imported `http::` to fully-qualified `k8s_openapi::http::`